### PR TITLE
Servo-Animate

### PR DESCRIPTION
**Pull request by dtex**
* Update Servo to use the ```Animate()``` class for ```Servo.to()```
* Stop emitting move:complete when the calling ```servo.to(n)``` and n = last
* Move temporal based Servo test scripts to extended folder

Please note: running ```grunt nodeunit:complete``` will throw an error during the extended servo tests. This error is coming from board.js (see #856)

### CODE REVIEW

**Comment by dtex** [comment]
Also, these changes are showing up on an earlier commit so I made a mistake at some point. Probably ran ```push --force``` when I shouldn't have. I'll get that fixed.

**Comment by dtex** [comment]
That commit problem is fixed, so this PR is ready for real

File: lib/servo.js [hunk 1]
```diff
@@ -254,87 +253,75 @@ util.inherits(Servo, Emitter);
...
-    if (distance < rate) {
-      rate = distance;
-    }
+    degrees += this.offset;
+    this.value = degrees;
 
-    if (this.interval) {
-      clearInterval(this.interval);
-    }
+    var history = priv.get(this).history;
```
Comment (rwaldron) [inline lib/servo.js]: This is already available as `state.history`

File: lib/servo.js [hunk 2]
```diff
@@ -448,72 +435,23 @@ Servo.prototype.center = function(time, rate) {
...
-    interval = 100,
-    step = 10;
 
-  opts = opts || {};
+  var options = {
+    keyFrames: [{degrees: this.range[0]}, {degrees: this.range[1]}],
+    metronomic: true,
+    loop: true,
+    easing: "inOutSine"
+  };
```
Comment (rwaldron) [inline lib/servo.js]: Nice!

File: lib/servo.js [hunk 3]
```diff
@@ -448,72 +435,23 @@ Servo.prototype.center = function(time, rate) {
...
-        process.nextTick(this.emit.bind(this, "sweep:full"));
-        step = abs;
-      }
-    }
-
-    degrees += step;
-
-    this.to(degrees);
-  }.bind(this), interval);
+  this.to(options);
```
Comment (rwaldron) [inline lib/servo.js]: :clap:

File: lib/servo.js (hunk 1 above)
Comment (dtex) [inline lib/servo.js]: Thanks. Fixed

**Comment by dtex** [comment]
The ```grunt nodeunit:complete``` tests are all clearing properly now via #862

**Comment by rwaldron** [comment]
LGTM

**Comment by dtex** [comment]
@makenai Is your delta bot in an easy to fire-up and test state? I'd like to know how this PR works on there and if it solves the problem you were experiencing.

**Comment by makenai** [comment]
@dtex I have ready to go! Unfortunately, it's 5000 miles away from where I am until the end of the month. :( The circuit is the one in the JS Robotics book though.

**Comment by rwaldron** [comment]
This will need a rebasing

**Comment by dtex** [comment]
Rebased

**Comment by rwaldron** [comment]
Are you just waiting on more reality checks?

**Comment by dtex** [comment]
Yes. I really wanted to see if this solved @makenai's Deltabot woes but he's a traveling man right now. I need to scrounge up parts and try it out myself.

**Comment by dtex** [comment]
I was able to replicate @makenai 's issues and can confirm that this PR solved the problem.

**Comment by rwaldron** [comment]
Land when ready (be sure to rebase)

**Comment by dtex** [comment]
Rebased and merged

**Pull request merged**